validphys2/src/validphys/config.py

```diff
@@ -1504,6 +1502,11 @@ def produce_group_dataset_inputs_by_metadata(
             {"data_input": NSList(group, nskey="dataset_input"), "group_name": name}
             for name, group in res.items()
         ]
+    
+    def produce_group_dataset_inputs_by_fitting_group(self, data_input, theory_covmat_flag):
+        if theory_covmat_flag is True:
+            return self.produce_group_dataset_inputs_by_metadata(data_input, "custom_group")
```

Comment:
We need to define a new grouping that just takes everything (if only to avoid the `unset` flag).

A possibility would be to use here `BIGEXP` or `ALL` and then in
https://github.com/NNPDF/nnpdf/blob/7b0ea262b48029e51fcd21a6a7ab83a3a5e6fdba/validphys2/src/validphys/config.py#L1475

change it so that if it receives `ALL` it doesn't look into the metadata and just groups everything together with `name=ALL`.

Reply:
Do you know where the groups are defined?

Reply:
The group themselves in the metadata in the plotting files. But I don't think this should be a group (since it should use all datasets) but rather a special case in `produce_group_dataset_inputs_by_metadata`.

Reply:
Ok but I believe that the return of produce_group_dataset_inputs_by_metadata is still expecting one group over which iterate. So what I can do is to call this group "ALL" or something like that (at least to avoid the name to be "unset"). Is this what you mean?

Reply:
Yes exactly!
Instead of using the `custom_group` (which is a group and doesn't necessarily mean all datasets).